Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -948,8 +953,25 @@ namespace OpenInfraPlatform
 				}
 
 				// direction and length of extrusion
-				const double depth = extrudedArea->Depth * this->UnitConvert()->getLengthInMeterFactor();
-				carve::geom::vector<3> extrusionVector = placementConverter->convertIfcDirection(extrudedArea->ExtrudedDirection);
+				const double depth = (typename IfcEntityTypesT::IfcLengthMeasure)(extrudedArea->Depth) * UnitConvert()->getLengthInMeterFactor();

```

Comment:
Is the cast to `IfcLengthMeasure` really needed?

Reply:
Resolved: 392efce